test/units/cli/test_galaxy.py

```diff
@@ -444,3 +450,123 @@ def test_template_ignore_similar_folder(self):
 
     def test_skeleton_option(self):
         self.assertEquals(self.role_skeleton_path, context.CLIARGS['role_skeleton'], msg='Skeleton path was not parsed properly from the command line')
+
+
+@pytest.fixture()
+def collection_skeleton(request):
+    name, skeleton_path = request.param
+
+    galaxy_args = ['ansible-galaxy', 'collection', 'init', '-c']
+
+    if skeleton_path is not None:
+        galaxy_args += ['--collection-skeleton', skeleton_path]
+
+    test_dir = tempfile.mkdtemp()
+    if not os.path.isdir(test_dir):
+        os.makedirs(test_dir)
+    galaxy_args += ['--init-path', test_dir, name]
+
+    gc = GalaxyCLI(args=galaxy_args)
+    gc.run()
+
+    namespace_name, collection_name = name.split('.', 1)
+    collection_dir = os.path.join(test_dir, namespace_name, collection_name)
+    yield collection_dir
+
+    if os.path.isdir(test_dir):
```

Comment:
This is not needed with `tmp_path_factory` because of tmp files are scoped to test run and are being garbage collected by `pytest` itself.
```suggestion
```